### ISSUE TITLE
TypeError: null is not an object (evaluating 'tabBar.items.count')

### ISSUE THREAD

**Issue by onury** [NativeScript/NativeScript#3568]
### Did you verify this is a real problem by searching [Stack Overflow](http://stackoverflow.com/questions/tagged/nativescript) and the [other open issues in this repo](https://github.com/NativeScript/nativescript/issues)?
Yes.

### Tell us about the problem
Just a regular `TabView` control, throws `TypeError: null is not an object (evaluating 'tabBar.items.count') getNativeFontValue@file: ...`.
```
CONSOLE ERROR file:///app/tns_modules/@angular/core/bundles/core.umd.js:3491:32: EXCEPTION: Uncaught (in promise): TypeError: null is not an object (evaluating 'tabBar.items.count')
CONSOLE ERROR file:///app/tns_modules/@angular/core/bundles/core.umd.js:3496:36: ORIGINAL STACKTRACE:
CONSOLE ERROR file:///app/tns_modules/@angular/core/bundles/core.umd.js:3497:36: resolvePromise@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:416:40
	resolvePromise@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:401:31
	file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:449:31
	invokeTask@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:223:42
	onInvokeTask@file:///app/tns_modules/@angular/core/bundles/core.umd.js:4396:51
	invokeTask@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:222:54
	runTask@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:123:57
	drainMicroTaskQueue@file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:355:42
	promiseReactionJob@[native code]
	UIApplicationMain@[native code]
	start@file:///app/tns_modules/tns-core-modules/application/application.js:251:26
	bootstrapApp@file:///app/tns_modules/nativescript-angular/platform-common.js:67:28
	bootstrapModule@file:///app/tns_modules/nativescript-angular/platform-common.js:55:26
	anonymous@file:///app/main.js:7:57
	evaluate@[native code]
	moduleEvaluation@[native code]
	[native code]
	promiseReactionJob@[native code]
CONSOLE ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:342:26: Unhandled Promise rejection: null is not an object (evaluating 'tabBar.items.count') ; Zone: angular ; Task: Promise.then ; Value: TypeError: null is not an object (evaluating 'tabBar.items.count') getNativeFontValue@file:///app/tns_modules/tns-core-modules/ui/tab-view/tab-view.js:427:25
```

### Which platform(s) does your issue occur on?
iOS

### Please provide the following version numbers that your issue occurs with:
- CLI:  2.5.0
- Cross-platform modules: 2.5.0
- Runtime(s): tns-ios 2.5.0

### Please tell us how to recreate the issue in as much detail as possible.
Just using a simple `TabView` like below.

### Is there code involved? If so, please share the minimal amount of code needed to recreate the problem.
```html
<TabView #tabView>
    <StackLayout *tabItem="{ title: 'Profile', iconSource: 'res://tab-profile' }"></StackLayout>
    <StackLayout *tabItem="{ title: 'List', iconSource: 'res://tab-list' }"></StackLayout>
    <StackLayout *tabItem="{ title: 'Other', iconSource: 'res://tab-other' }"></StackLayout>
</TabView>
```

I think `if (!tabBar || !tabBar.items) return;` should be added @ after [tab-view.ios.ts#L499](https://github.com/NativeScript/NativeScript/blob/master/tns-core-modules/ui/tab-view/tab-view.ios.ts#L499) to solve this.

**Comment by tsonevn** [comment]
Hi @onury,
This issue has been related with [#654](https://github.com/NativeScript/nativescript-angular/issues/654).

This problem has been reproduced while setting `fontSize` and `fontFamily` to the TabView component in `NativeScript Angular 2` project. 

This behavior has been reproduced only on `iOS`.

Please keep track on the other issue for further info.

**Comment by onury** [comment]
Hi @tsonevn,
I removed all styling from `TabView` but I still get a similar error, just without the `getNativeFontValue`
```
ERROR file:///app/tns_modules/nativescript-angular/zone-js/dist/zone-nativescript.js:344:22: Error: Uncaught (in promise): TypeError: null is not an object (evaluating 'tabBar.items.count')
```

**Comment by tsonevn** [comment]
Hi @onury,

Could you provide some more info about your problem? It would help if you could give us sample project, which could be debugged locally?

Regards,
@tsonevn

**Comment by onury** [comment]
@tsonevn Currently I cannot re-produce this with a clean project. Only happens in my main project. Still investigating..

But If you take a look at the PR, it's very logical to check for whether `tabBar` or `tabBar.items` are defined/not-null before iterating (just like the other 2 instance methods that check for `this.items`, such as `_updateIOSTabBarTextTransform()` and `_updateIOSTabBarColorsAndFonts()`  iterating over `tabBar.items` in the same file).

I can confirm that [this commit](https://github.com/onury/NativeScript/commit/5679e78507df9611034b8a62a6763457389201cd) resolves the issue in my main project.

**Comment by onury** [comment]
Ok. As noted in #654 removing mentioned styling from `TabView` is not enough. I also had to remove `font-family` from `Page`. (In other words, I can re-produce this in  a clean project if I add styles to `Page`).

The problem is setting a font related style to `TabView` or `Page` (which is inherited by `TabView`) immediately triggers `getNativeFontValue()` method; before `tabBar` or it's items are initialized. 

And since `getNativeFontValue()` method doesn't check whether `tabBar` or `tabBar.items` collection is available; it throws on `tabBar.items.count`.

My PR resolves this and #654. That way you can apply styles to `TabView` and/or `Page` and it'll work.

**Comment by tsonevn** [comment]
Hi @onury,
Thank you for your PR.

We will review it and if all tests passed, we will merge it.

Regards,
@tsonevn

**Comment by pap5508** [comment]
@tsonevn : does this issue is resolved..??

**Comment by tsonevn** [comment]
Hi @pap5508,
This issue has been resolved with PR [#3587](https://github.com/NativeScript/NativeScript/pull/3587) and the fix will be available in the next NativeScript release.

**Comment by vchimev** [comment]
Fixed with https://github.com/NativeScript/NativeScript/pull/3587 in 2.5.1.
